src/Tickets/Commerce/Payments_Tab.php

```diff
@@ -106,4 +117,99 @@ class="tec-tickets__admin-settings-tickets-commerce-toggle-checkbox tribe-depend
 		 */
 		return apply_filters( 'tec_tickets_commerce_settings_top_level', $top_level_settings );
 	}
+
+	/**
+	 * Generate Chceckout and Success page if not found.
+	 *
+	 * @since TBD
+	 */
+	public function maybe_generate_pages() {
+
+		$tc_enabled = tribe_get_request_var( Tickets_Settings::$tickets_commerce_enabled );
+
+		if ( ! tribe_is_truthy( $tc_enabled ) ) {
+			return;
+		}
+
+		$this->maybe_auto_generate_checkout_page();
+		$this->maybe_auto_generate_order_success_page();
+	}
+
+	/**
+	 * Generate Checkout page with shortcode if not available.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function maybe_auto_generate_checkout_page() {
+		if ( tribe( Checkout::class )->page_has_shortcode() ) {
+			return false;
+		}
+
+		$page_slug   = 'tickets-checkout';
+		$page_exists = get_page_by_path( $page_slug );
+
+		if ( $page_exists ) {
+			return false;
+		}
+
+		$page_data = [
+			'post_status'    => 'publish',
+			'post_type'      => 'page',
+			'post_author'    => 1,
+			'post_name'      => $page_slug,
+			'post_title'     => 'Tickets Checkout',
```

Comment:
We may want to have that in gettext functions so that it also works in other languages (if translated).